scrape.py

```diff
@@ -120,33 +120,63 @@ def convert_thread(start: int, end: int):
         print(f"[{threading.current_thread().name}] Converting page {i} to PDF")
         convert_html_to_pdf(i)
 
-
-assert(NUM_PAGES % NUM_THREADS == 0)
 chunk_size = int(NUM_PAGES / NUM_THREADS)
 
 # Download threads
 threads: list[threading.Thread] = []
 start = 1
-for i in range(0, NUM_THREADS):
-    thread = threading.Thread(target=download_thread, args=(start, start + chunk_size))
+
+if(NUM_THREADS > 1):
```

Comment:
Sorry. Let me clarify. This if statement should only wrap the for loop, not the code below the for loop. For example:

```py
if (NUM_THREADS > 1):
    for i in range(0, NUM_THREADS - 1):
        pass

thread = threading. Thread...
```

Do this for both of the loops for download and convert and then we don't need the extra code wrapped by the ``if(NUM_THREADS == 1)``